smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ShapeValueGenerator.kt

```diff
@@ -138,6 +138,12 @@ class ShapeValueGenerator(
     ) : NodeVisitor<Unit> {
 
         override fun objectNode(node: ObjectNode) {
+            if (currShape.type === ShapeType.DOCUMENT) {
```

Comment:
**Question**: Why `===`? Won't `==` be sufficient here?

Reply:
You have witnessed firsthand my javascript reflex. I didn't even realize I did this!